api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -45,7 +46,7 @@ static StructType getType(StructType partitionType) {
         required(102, "partition", partitionType),
         required(103, "record_count", LongType.get()),
         required(104, "file_size_in_bytes", LongType.get()),
-        required(105, "block_size_in_bytes", LongType.get()),
+        optional(105, "block_size_in_bytes", LongType.get()),
```

Comment:
Could you add a note that although it is optional, we need to always write it for older readers?

Reply:
Done